lang/racket/Portfile

```diff
@@ -34,10 +35,47 @@ patch.dir           ${workpath}/${distname}
 # diff -NaurdwB -I '^;;[[:space:]]*' ./racket-orig/ ./racket-new | sed -E -e 's/\.\/racket-(orig|new)/\./g' > ~/Downloads/paths-to-openssl.diff
 patchfiles-append   paths-to-openssl.diff
 
+# please remove after these commits are released
+# https://github.com/racket/racket/commit/85ecfcb873092773a4897d68ddcb9bed7f941497
+# https://github.com/racket/gui/commit/aea05e8b91bb78a398374f2b32f4f13a20303118#diff-7f2aa32a6df5d6cb5a9c21dd8fb9f00bd480fb38715269eeef4411676dda783a
+# diff -NaurdwB -I '^;;[[:space:]]*' ./racket-orig/ ./racket-new | sed -E -e 's/\.\/racket-(orig|new)/\./g' > ~/Downloads/patch-destdir-codesign-uninit.diff
+patchfiles-append   patch-destdir-codesign-uninit.diff
+                   
 post-patch {
     foreach rkt {libcrypto.rkt libssl.rkt} {
         reinplace "s|@@PREFIX@@|${prefix}|g" \
-                    ${worksrcpath}/../collects/openssl/${rkt}
+            ${worksrcpath}/../collects/openssl/${rkt}
+    }
+
+    # https://github.com/racket/racket/issues/5055#issuecomment-2321445579
+    # collects/racket/private/so-search.rkt
+    reinplace -E \
+        "s|^#hash\\((.+)\\)$|#hash(\\1 (lib-search-dirs . (\"${prefix}/lib\" \"${frameworks_dir}\" #f)))|" \
+        ${worksrcpath}/../etc/config.rktd
+
+    # use MacPorts-installed libraries
```

Comment:
all this manual changing if library names seems very fragile and it will need work every time a library version is changed - is there no other way?

Reply:
I made this a little less fragile by using version wildcards for the sources.